src/controllers/userProfileController.js

```diff
@@ -763,6 +768,80 @@ const userProfileController = function (UserProfile) {
     res.status(200).send({ refreshToken: currentRefreshToken });
   };
 
+  const getInfoCollections = function (req, res) {
+    UserProfile.find({}, 'infoCollections')
+      .then((users) => {
+        const infoCollections = users.map((user) => user.infoCollections);
+        res.status(200).json(infoCollections);
+      })
+      .catch((error) => {
+        res.status(500).send({ error: error.message });
+      });
+  };
+  
+  const putInfoCollections = function (req, res) {
```

Comment:
I noticed that you're using `UserProfile.updateMany({}, { $set: { infoCollections } })` to update the `infoCollections` field for all users. I would suggest considering including the user's ID and their corresponding `infoCollections` in the request body, and then iterating over these users and `infoCollections` in the method to update the `infoCollections` for each user. 

```javascript
const { userInfos } = req.body; // userInfos should be an array of { userId, infoCollections }
for (let userInfo of userInfos) {
  await UserProfile.updateOne({ _id: userInfo.userId }, { $set: { infoCollections: userInfo.infoCollections } });
}